ctk/common/src/main/kotlin/org/codice/compliance/verification/core/SubjectComparisonVerifier.kt

```diff
@@ -66,30 +68,49 @@ class SubjectComparisonVerifier(private val samlResponseDom: Node,
         if (subjectSet.size < 2) return
 
         Sets.combinations(subjectSet, 2).forEach {
-            verifyIdContentsMatchSSO(it.first(), it.last())
+            verifyIdContentsMatch(it.first(), it.last(), SAMLProfiles_4_1_4_2_d)
         }
     }
 
     /**
      * Compares the text content of the identifiers. Special checking due to nameIdPolicyFormat.
      */
-    private fun verifyIdContentsMatchSSO(id1: Node, id2: Node) {
+    private fun verifyIdContentsMatch(id1: Node, id2: Node, samlCode: SAMLSpecRefMessage) {
         val format1 = id1.filteredFormatValue
         val format2 = id2.filteredFormatValue
 
         // If they share a format value that isn't "unspecified", but have different contents
         if (format1 == format2
                 && format1 != null
                 && id1.textContent != id2.textContent) {
-            throw SAMLComplianceException.create(SAMLProfiles_4_1_4_2_d,
-                    SAMLCore_3_3_4_b,
-                    message = "Two Response Subject identifiers have identical Format attributes " +
+            throw SAMLComplianceException.create(SAMLCore_3_3_4_b,
+                    samlCode,
+                    message = "The identifiers have identical Format attributes " +
                             "[$format1], but the content of one [${id1.textContent}] is not " +
                             "equal to the content of the other [${id2.textContent}]",
                     node = samlResponseDom)
         }
     }
 
+    /**
+     * This function verifies that the identifier in the LogoutRequest strongly matches
+     * the identifier from the SAML assertion that was issued when that principal was logged in.
+     */
+    fun verifyIdsMatchSLO(logoutRequest: Node) {
+        val assertionId =
+                samlResponseDom.recursiveChildren(ASSERTION).firstOrNull()?.children(SUBJECT)
+                        ?.firstOrNull()?.id ?: throw IllegalArgumentException(
+                        "Could not find the assertion's identifier on the response.")
+
+        val logoutRequestId = logoutRequest.children().firstOrNull {
+            it.localName == NAME_ID || it.localName == BASE_ID
+        } ?: throw IllegalArgumentException(
+                "Could not find the logout request's identifier.")
+
+//        verifyIdAttributesMatch(assertionId, logoutRequestId, SAMLProfiles_4_4_4_1_c)
```

Comment:
❓ Is this commented out because DDF does it incorrectly?

Reply:
Yup. I'll add a comment on this line and create a ticket for it.